pkg/chunk/chunk_store.go

```diff
@@ -151,6 +151,12 @@ func (c *Store) calculateDynamoWrites(userID string, chunks []Chunk) (WriteBatch
 
 // Get implements ChunkStore
 func (c *Store) Get(ctx context.Context, from, through model.Time, allMatchers ...*metric.LabelMatcher) ([]local.SeriesIterator, error) {
+	now := model.Now()
+	if through.After(now) {
+		util.WithContext(ctx).Warnf("Capping TimeSpanHi from future=%v to now=%v", through, now)
```

Comment:
Do we need to log these? They may be happening frequently (with dashboards etc) and it is only "parsing" the input to remove the need to query data which cannot exist.